pyiceberg/catalog/sql.py

```diff
@@ -111,7 +111,7 @@ def _ensure_tables_exist(self) -> None:
                 stmt = select(1).select_from(table)
                 try:
                     session.scalar(stmt)
-                except OperationalError:
+                except (OperationalError, ProgrammingError): # sqlalchemy returns OperationalError in case of sqlite and ProgrammingError with postgres.
```

Comment:
```suggestion
                except (
                    OperationalError,
                    ProgrammingError,
                ):  # sqlalchemy returns OperationalError in case of sqlite and ProgrammingError with postgres.
```